optas/builder.py

```diff
@@ -75,20 +75,22 @@ def __init__(self, T, robots=[], tasks=[], optimize_time=False, derivs_align=Fal
 
         # Setup decision variables
         self._decision_variables = SXContainer()
+        self._parameters = SXContainer()
         for model in self._models:
             for d in model.time_derivs:
-                n = model.state_name(d)
+                n_s = model.state_name(d)
+                n_p = model.parameter_name(d)
                 if model.T is None:
                     t = T-d if not derivs_align else T
                 else:
                     t = model.T
-                self.add_decision_variables(n, model.dim, t)
+                self.add_decision_variables(n_s, len(model.opt_joint_names), t)
+                self.add_parameter(n_p, len(model.param_joint_names), t)
```

Comment:
This will break when the `model` is a `TaskModel`. We need to think about how `TaskModel` handles parameters.

Reply:
Added if condition to use this only if RobotModel and do the previous thing if TaskModel.
Does that solve it?
I don't know how to test it for the case of TaskModel!